### DISCUSSION TITLE
Integration Tests for Next JS api with middleware

### DISCUSSION THREAD

**Discussion by Annada9** [vercel/next.js#12762] [Help]
I want to write Integration tests for Next JS api. My Next JS api is wrapped with middleware handler which authenticates the user. Can anyone please suggest how to achieve the same?

**Comment by craigmulligan** [comment]
Hey @Annada9 it's difficult to say without looking at the code, but you could always export you handler without the middleware and then test them independently. 

Here's an example: https://github.com/hobochild/boiler/blob/master/__tests__/pages/api/logout.test.js

You'll see line 22-25 add the session to the request and "login" the user manually rather than using the middleware.
